src/Common/src/Interop/Unix/System.Native/Interop.ReadDir.cs

```diff
@@ -50,14 +50,17 @@ internal ReadOnlySpan<char> GetName(Span<char> buffer)
 
                 int charCount = Encoding.UTF8.GetChars(nameBytes, buffer);
                 ReadOnlySpan<char> value = buffer.Slice(0, charCount);
-                Debug.Assert(value.IndexOf('\0') == -1, "should not have embedded nulls");
+                Debug.Assert(NameLength != -1 || value.IndexOf('\0') == -1, "should not have embedded nulls if we parsed the end of string");
                 return value;
             }
         }
 
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_OpenDir", SetLastError = true)]
         internal static extern SafeDirectoryHandle OpenDir(string path);
 
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_OpenDir", SetLastError = true)]
```

Comment:
We still use the other one somewhere?

Reply:
I looked again, the other usage is coming from the other copy (mirror from CoreLib)